### ISSUE TITLE
Mapbox' GeolocateControl does not trigger onViewportChange

### ISSUE THREAD

**Issue by felixerdy** [uber/react-map-gl#672]
I implemented Mapbox' GeolocateControl like this: 
```js
componentDidMount() {
    this.map.getMap().addControl(new mapboxgl.GeolocateControl({
      positionOptions: {
        enableHighAccuracy: true,
      },
      trackUserLocation: true,
    }))
}
```

where `this.map` is the reference to `ReactMapGL`. 
```js
<ReactMapGL
          ref={el => (this.map = el)}
          {...this.props.viewport}
          onViewportChange={viewport => this.props.setViewport(viewport)}
...
```

It all works fine and I can visualize the users location. The transition to the users location however does not trigger `onViewportChange`. Panning / zooming by mouse works and triggers `onViewportChange`of course.

**Comment by Pessimistress** [comment]
Unfortunately, Mapbox's native controls do not trigger any React callbacks. We can potentially add a new React wrapper for this control, hopefully when I have time to work on this.

Relevant ticket: https://github.com/uber/react-map-gl/issues/484

**Comment by Pessimistress** [comment]
An React version of GeolocateControl is included in v4.1.
